contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -598,6 +601,7 @@ contract GoodGhostingPolygonCurve is Ownable, Pausable {
         } else {
             // handling impermanent loss case
             impermanentLossShare = (totalBalance.mul(uint(100))).div(totalGamePrincipal);
+            originalTotalGamePrincipal = int(totalGamePrincipal - totalBalance);
```

Comment:
we don't need this. We always want `originalTotalGamePrincipal` to store the original value of the principal, not the difference between them. So we can just get rid of this, and only keep the code added above (`originalTotalGamePrincipal = int(totalGamePrincipal);`)